app/models/osf-model.ts

```diff
@@ -173,9 +173,21 @@ export default class OsfModel extends Model {
             if (response.data.relationships && apiRelationshipName in response.data.relationships) {
                 const relationship = response.data.relationships[apiRelationshipName];
                 if ('links' in relationship) {
-                    set(this.relatedCounts, relationshipName, relationship.links.related.meta.count);
+                    if (typeof relationship.links.related === 'object') {
+                        if (relationship.links.related.meta) {
+                            if (typeof relationship.links.related.meta.count === 'number') {
+                                set(this.relatedCounts, relationshipName, relationship.links.related.meta.count);
+                            } else {
+                                throw new Error(`Count not found in related link meta ${errorContext}`);
+                            }
+                        } else {
+                            throw new Error(`Meta not found in related link ${errorContext}`);
+                        }
+                    } else {
+                        throw new Error(`Related link not found in relationship links ${errorContext}`);
+                    }
                 } else {
-                    throw new Error(`Relationship link not found ${errorContext}`);
+                    throw new Error(`Relationship links not found ${errorContext}`);
```

Comment:
Not sure how I feel about this pattern... I don't like the nesting and repetition, but I do like the useful error messages. What about something like
```ts
try {
    relatedCount = relationship.links.related.meta.count;
} catch {
    throw new Error('Could not find related count ${errorContext}');
}
```
The message is descriptive enough to point you in the right direction without drilling down excessively.

Reply:
All the ifs are needed because TypeScript.

Reply:
I guess what I want is https://github.com/tc39/proposal-optional-chaining

This is fine until then :+1:

Reply:
Yeah, that would be awesome.

Reply:
Good read with a temporary solution: https://medium.com/inside-rimeto/optional-chaining-in-typescript-622c3121f99b